### DISCUSSION TITLE
How can I determine the font size of an image I'm creating?

### DISCUSSION THREAD

**Discussion by Pival81** [dlemstra/Magick.NET#801] [Help]
This is my code: 
```
var drummerImage = new MagickImage();
drummerImage.Read($"caption:{Meme.DrummerText}", new MagickReadSettings()
{
   BackgroundColor = MagickColors.Transparent,
   FillColor = MagickColors.White,
   Font = "Impact",
   Width = 700,
   Height = 460,
   TextGravity = Gravity.Center
});
drummerImage.Write($"./Temp/drummer_{Meme.Guid}.png");
```
Basically I'm creating images containing text, and I'm relying on ImageMagick to set the font's size based on the size of the image. It's doing a wonderful job, too good even. 
In fact, sometimes the string I'm using is very short, and the image ends up ridiculously big.
Since I'd like to avoid changing the size of the image according to the string, I thought of setting a maximum font size, but I can't seem to be able to do that. Is there any way of even seeing the font size that ImageMagick is calculating when writing the image?

**Comment by Pival81** [comment]
Alright, so, I ended up figuring out how to do this myself, it was pretty simple but it's sort of hacky, if someone knows of a better way to do this please let me know.
```
var catImage = new MagickImage();
catImage.Read($"caption:{Meme.CatText}", new MagickReadSettings()
{
    BackgroundColor = MagickColors.Transparent,
    FillColor = MagickColors.Black,
    Font = "Impact",
    Width = 800,
    Height = 800,
    TextGravity = Gravity.Center
});
if (Int32.Parse(catImage.GetAttribute("caption:pointsize")) > 120)
{
    catImage.Settings.FontPointsize = 120;
    catImage.Read($"caption:{Meme.CatText}", catImage.Settings as MagickReadSettings);
}
catImage.Write($"./Temp/cat_{Meme.Guid}.png");
```

So basically there's this method `GetAttribute` that lets me get the attributes of an image, with this I was able to retrieve the font point size of the text I'm generating before I'm actually generating it.
I determined that 120 - at least for that particular image - was the max point size I wanted, so I'm just checking everytime that if it exceeds that number, then I want to use the same settings but with a fixed point sized, instead of letting Magick.NET calculate it. As I said, it's very hacky and probably quite inefficient, since I'm calling `Read` two times, and I'm also reusing the same `Settings` object by casting it to `MagickReadSettings`, but hey, for now it gets the job done.
I'll be honest, I was ecstatic when I saw it was working as intended, I spent most of the day on this.

**Reply by dlemstra** [reply]
You can limit the font size by setting the attribute `caption:max-pointsize`. If you set that before reading the image the font size will not be higher than the limit you set.

**Reply by Pival81** [reply]
Perhaps I'm doing something wrong, but it doesn't seem to be working:
```
using (var drumImage = new MagickImage())
{
    drumImage.SetAttribute("caption:max-pointsize", "90");
    drumImage.Read($"caption:{Meme.DrumText}", new MagickReadSettings()
    {
        BackgroundColor = MagickColors.Transparent,
        FillColor = MagickColors.White,
        Font = "Impact",
        Width = 1000,
        Height = 300,
        TextGravity = Gravity.Center
    });
    drumImage.Write($"./Temp/drum_{Meme.Guid}.png");
}
```
I also tried setting the attribute after the Read mehod, but it doesn't work either.

**Reply by dlemstra** [reply]
I am sorry for using the word attribute that pointed you in the wrong direction. You will need to do this instead:

```C#
var readSettings = new MagickReadSettings()
{
    BackgroundColor = MagickColors.Transparent,
    FillColor = MagickColors.White,
    Font = "Impact",
    Width = 1000,
    Height = 300,
    TextGravity = Gravity.Center
};
readSettings.SetDefine(MagickFormat.Caption, "max-pointsize", "90");

// You can do this in the next version:
readSettings.SetDefine(MagickFormat.Caption, "max-pointsize", 90);

// And you can also do this:
var readSettings = new MagickReadSettings()
{
    BackgroundColor = MagickColors.Transparent,
    FillColor = MagickColors.White,
    Font = "Impact",
    Width = 1000,
    Height = 300,
    TextGravity = Gravity.Center,
    Defines = new CaptionReadDefines
    {
        MaxFontPointsize = 90
    }
};

drumImage.Read($"caption:TEST", readSettings);
```
